Content.Server/Fluids/EntitySystems/PuddleSystem.Spillable.cs

```diff
@@ -92,13 +92,14 @@ private void SplashOnMeleeHit(EntityUid uid, SpillableComponent component, Melee
                 continue;
             }
 
-            var splitSolution = _solutionContainerSystem.SplitSolution(uid, solution, totalSplit / hitCount);
+            var initialVolume = solution.Volume;
+            SpillSolutionOnTarget(solution, hit, (totalSplit / (solution.Volume * hitCount)).Float());
+            var volumeAfterSpill = solution.Volume;
```

Comment:
This should be using FixedPoint2 not floats as it will creative / remove solution due to rounding.

Reply:
`SpillSolutionOnTarget` takes fraction of what to take, rounding is up to event subscribers